tests/unit_tests/core/router/identity.cc

```diff
@@ -85,4 +85,20 @@ BOOST_AUTO_TEST_CASE(ParseIdentityFailure)
         0);
 }
 
+BOOST_AUTO_TEST_CASE(CreateValidRoutingKey)
+{
+  core::IdentityEx ident;
+  BOOST_CHECK(
+      ident.FromBuffer(m_AliceIdentity.data(), m_AliceIdentity.size()));
+  BOOST_CHECK_NO_THROW(core::CreateRoutingKey(ident.GetIdentHash()));
+  BOOST_CHECK(!core::CreateRoutingKey(ident.GetIdentHash()).IsZero());
+}
+
+BOOST_AUTO_TEST_CASE(CreateInvalidRoutingKey)
+{
+  kovri::core::IdentHash zeroed_ident;
+  BOOST_CHECK_THROW(core::CreateRoutingKey(nullptr), std::invalid_argument);
+  BOOST_CHECK(core::CreateRoutingKey(zeroed_ident).IsZero());
+}
+
```

Comment:
This test-case needs to ensure proper / spec-defined date formatting.

Reply:
How can we inspect the date formatting without re-implementing the function, or is that what you mean? I'm confused...

Reply:
>How can we inspect the date formatting without re-implementing the function, or is that what you mean?

That would be expected if you wish to adequately test the data before it is hashed. Perhaps split into multiple functions (formatting versus key creation / hashing)?